File: internal/consts/backend.go
package consts

type AuthProvider string

const (
	AuthProviderEmail  AuthProvider = "email"
	AuthProviderGoogle AuthProvider = "google"
	AuthProviderGithub AuthProvider = "github"
)

// Organization Types
type OrganizationType string

const (
	OrgTypeInternal OrganizationType = "internal"
	OrgTypeClient   OrganizationType = "client"
	OrgTypeProvider OrganizationType = "provider"
)

// User Roles
type UserRole string

const (
	RoleInternalSU      UserRole = "internal_su"
	RoleInternalAdmin   UserRole = "internal_admin"
	RoleInternalSupport UserRole = "internal_support"
	RoleClientAdmin     UserRole = "client_admin"
	RoleDER             UserRole = "der"
	RoleSafetyManager   UserRole = "safety_manager"
	RoleHRManager       UserRole = "hr_manager"
	RoleEmployee        UserRole = "employee"
	RoleMRO             UserRole = "mro"
	RoleSAP             UserRole = "sap"
	RoleMedicalExaminer UserRole = "medical_examiner"
)

// Subscription Tiers
type SubscriptionTier string

const (
	TierStarter      SubscriptionTier = "starter"
	TierProfessional SubscriptionTier = "professional"
	TierEnterprise   SubscriptionTier = "enterprise"
	TierCustom       SubscriptionTier = "custom"
)

// Subscription Status
type SubscriptionStatus string

const (
	SubStatusActive    SubscriptionStatus = "active"
	SubStatusCancelled SubscriptionStatus = "cancelled"
	SubStatusSuspended SubscriptionStatus = "suspended"
	SubStatusExpired   SubscriptionStatus = "expired"
	SubStatusPending   SubscriptionStatus = "pending"
)

// Test Types
type TestType string

const (
	TestTypePreEmployment       TestType = "pre_employment"
	TestTypeRandom              TestType = "random"
	TestTypePostAccident        TestType = "post_accident"
	TestTypeReasonableSuspicion TestType = "reasonable_suspicion"
	TestTypeReturnToDuty        TestType = "return_to_duty"
	TestTypeFollowUp            TestType = "follow_up"
)

// Test Categories
type TestCategory string

const (
	TestCategoryDrug        TestCategory = "drug"
	TestCategoryAlcohol     TestCategory = "alcohol"
	TestCategoryDrugAlcohol TestCategory = "drug_alcohol"
)

// Test Status
type TestStatus string

const (
	TestStatusOrdered    TestStatus = "ordered"
	TestStatusScheduled  TestStatus = "scheduled"
	TestStatusInProgress TestStatus = "in_progress"
	TestStatusCompleted  TestStatus = "completed"
	TestStatusCancelled  TestStatus = "cancelled"
	TestStatusNoShow     TestStatus = "no_show"
)

// Test Results
type TestResult string

const (
	TestResultNegative    TestResult = "negative"
	TestResultPositive    TestResult = "positive"
	TestResultRefusal     TestResult = "refusal"
	TestResultAdulterated TestResult = "adulterated"
	TestResultSubstituted TestResult = "substituted"
	TestResultInvalid     TestResult = "invalid"
)

// MVR Status
type MVRStatus string

const (
	MVRStatusOrdered  MVRStatus = "ordered"
	MVRStatusReceived MVRStatus = "received"
	MVRStatusReviewed MVRStatus = "reviewed"
	MVRStatusFlagged  MVRStatus = "flagged"
)

// Physical Status
type PhysicalStatus string

const (
	PhysicalStatusScheduled     PhysicalStatus = "scheduled"
	PhysicalStatusCompleted     PhysicalStatus = "completed"
	PhysicalStatusFailed        PhysicalStatus = "failed"
	PhysicalStatusExpired       PhysicalStatus = "expired"
	PhysicalStatusPendingReview PhysicalStatus = "pending_review"
)

// Background Check Types
type BackgroundCheckType string

const (
	BackgroundCriminalHistory        BackgroundCheckType = "criminal_history"
	BackgroundEmploymentVerification BackgroundCheckType = "employment_verification"
	BackgroundReferenceCheck         BackgroundCheckType = "reference_check"
	BackgroundEducationVerification  BackgroundCheckType = "education_verification"
	BackgroundLicenseVerification    BackgroundCheckType = "license_verification"
	BackgroundCreditCheck            BackgroundCheckType = "credit_check"
)

// Background Check Status
type BackgroundCheckStatus string

const (
	BackgroundStatusOrdered        BackgroundCheckStatus = "ordered"
	BackgroundStatusInProgress     BackgroundCheckStatus = "in_progress"
	BackgroundStatusCompleted      BackgroundCheckStatus = "completed"
	BackgroundStatusRequiresReview BackgroundCheckStatus = "requires_review"
)

// Document Types
type DocumentType string

const (
	DocTypeTestResult          DocumentType = "test_result"
	DocTypeMVRReport           DocumentType = "mvr_report"
	DocTypeMedicalCertificate  DocumentType = "medical_certificate"
	DocTypeTrainingCertificate DocumentType = "training_certificate"
	DocTypeComplianceDocument  DocumentType = "compliance_document"
	DocTypePolicyDocument      DocumentType = "policy_document"
	DocTypeBackgroundReport    DocumentType = "background_report"
)

// Notification Types
type NotificationType string

const (
	NotificationEmail NotificationType = "email"
	NotificationSMS   NotificationType = "sms"
	NotificationInApp NotificationType = "in_app"
	NotificationPhone NotificationType = "phone"
)

// Notification Priorities
type NotificationPriority string

const (
	NotificationPriorityLow    NotificationPriority = "low"
	NotificationPriorityNormal NotificationPriority = "normal"
	NotificationPriorityHigh   NotificationPriority = "high"
	NotificationPriorityUrgent NotificationPriority = "urgent"
)

// Workflow Status
type WorkflowStatus string

const (
	WorkflowStatusPending   WorkflowStatus = "pending"
	WorkflowStatusRunning   WorkflowStatus = "running"
	WorkflowStatusCompleted WorkflowStatus = "completed"
	WorkflowStatusFailed    WorkflowStatus = "failed"
	WorkflowStatusCancelled WorkflowStatus = "cancelled"
)

type OnboardingStatus string

const (
	StatusPending            OnboardingStatus = "pending"
	StatusEmailVerified      OnboardingStatus = "email_verified"
	StatusSubscribed         OnboardingStatus = "subscribed"
	StatusOnboardingComplete OnboardingStatus = "complete"
	StatusFailed             OnboardingStatus = "failed"
)

File: v1-consortium-web-pkg/src/lib/types/backend.ts
// Organization and User Management Enums

export enum OrganizationType {
  INTERNAL = 'internal',
  CLIENT = 'client',
  PROVIDER = 'provider'
}

export enum UserRole {
  // Internal roles
  INTERNAL_SU = 'internal_su',
  INTERNAL_ADMIN = 'internal_admin',
  INTERNAL_SUPPORT = 'internal_support',
  
  // Client roles
  CLIENT_ADMIN = 'client_admin',
  DER = 'der',
  SAFETY_MANAGER = 'safety_manager',
  HR_MANAGER = 'hr_manager',
  EMPLOYEE = 'employee',
  
  // Provider roles
  MRO = 'mro',
  SAP = 'sap',
  MEDICAL_EXAMINER = 'medical_examiner'
}

// Subscription Management Enums

export enum SubscriptionTier {
  STARTER = 'starter',
  PROFESSIONAL = 'professional',
  ENTERPRISE = 'enterprise',
  CUSTOM = 'custom'
}

export enum SubscriptionStatus {
  ACTIVE = 'active',
  CANCELLED = 'cancelled',
  SUSPENDED = 'suspended',
  EXPIRED = 'expired',
  PENDING = 'pending'
}

// Testing and Compliance Enums

export enum TestType {
  PRE_EMPLOYMENT = 'pre_employment',
  RANDOM = 'random',
  POST_ACCIDENT = 'post_accident',
  REASONABLE_SUSPICION = 'reasonable_suspicion',
  RETURN_TO_DUTY = 'return_to_duty',
  FOLLOW_UP = 'follow_up'
}

export enum TestCategory {
  DRUG = 'drug',
  ALCOHOL = 'alcohol',
  DRUG_ALCOHOL = 'drug_alcohol'
}

export enum TestStatus {
  ORDERED = 'ordered',
  SCHEDULED = 'scheduled',
  IN_PROGRESS = 'in_progress',
  COMPLETED = 'completed',
  CANCELLED = 'cancelled',
  NO_SHOW = 'no_show'
}

export enum TestResult {
  NEGATIVE = 'negative',
  POSITIVE = 'positive',
  REFUSAL = 'refusal',
  ADULTERATED = 'adulterated',
  SUBSTITUTED = 'substituted',
  INVALID = 'invalid'
}

// Motor Vehicle Record Enums

export enum MVRStatus {
  ORDERED = 'ordered',
  RECEIVED = 'received',
  REVIEWED = 'reviewed',
  FLAGGED = 'flagged'
}

// Physical and Medical Enums

export enum PhysicalStatus {
  SCHEDULED = 'scheduled',
  COMPLETED = 'completed',
  FAILED = 'failed',
  EXPIRED = 'expired',
  PENDING_REVIEW = 'pending_review'
}

// Background Check Enums

export enum BackgroundCheckType {
  CRIMINAL_HISTORY = 'criminal_history',
  EMPLOYMENT_VERIFICATION = 'employment_verification',
  REFERENCE_CHECK = 'reference_check',
  EDUCATION_VERIFICATION = 'education_verification',
  LICENSE_VERIFICATION = 'license_verification',
  CREDIT_CHECK = 'credit_check'
}

export enum BackgroundCheckStatus {
  ORDERED = 'ordered',
  IN_PROGRESS = 'in_progress',
  COMPLETED = 'completed',
  REQUIRES_REVIEW = 'requires_review'
}

// Document Management Enums

export enum DocumentType {
  TEST_RESULT = 'test_result',
  MVR_REPORT = 'mvr_report',
  MEDICAL_CERTIFICATE = 'medical_certificate',
  TRAINING_CERTIFICATE = 'training_certificate',
  COMPLIANCE_DOCUMENT = 'compliance_document',
  POLICY_DOCUMENT = 'policy_document',
  BACKGROUND_REPORT = 'background_report'
}

// Notification Enums

export enum NotificationType {
  EMAIL = 'email',
  SMS = 'sms',
  IN_APP = 'in_app',
  PHONE = 'phone'
}

export enum NotificationPriority {
  LOW = 'low',
  NORMAL = 'normal',
  HIGH = 'high',
  URGENT = 'urgent'
}

// Workflow Management Enums

export enum WorkflowStatus {
  PENDING = 'pending',
  RUNNING = 'running',
  COMPLETED = 'completed',
  FAILED = 'failed',
  CANCELLED = 'cancelled'
}
